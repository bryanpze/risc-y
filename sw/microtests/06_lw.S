.text
start:
    lui x1, 1 // should match the address of e
    lw x2, 0(x1)
    lw x3, 4(x1)
    lw x4, 8(x1)
    lw x5, 12(x1)
    lw x6, -4(x1)
    lw x7, -8(x1)
    lw x8, -12(x1)
    lw x9, -16(x1)
    addi x10, x0, 0x400
    addi x10, x10, 0x400
    lw x11, 0x7F0(x10) // should matche the address of a
    lw x12, 0x7F4(x10)
    lw x13, 0x7F8(x10)
    lw x14, 0x7FC(x10)
    nop
    nop
    nop
    nop
    unimp

.data
. = 0x0FF0
a:      .word 0
b:      .word 1
c:      .word 52
d:      .word -215
e:      .word 17
f:      .word 59
g:      .word 12
h:      .word -16

