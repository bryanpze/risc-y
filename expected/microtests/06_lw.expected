Reached unsupported instruction
Total Clock Cycles =                   32
Dumping the state of the processor
pc = 0x0000004c
x0 = 0x00000000
x1 = 0x00001000
x2 = 0x00000000
x3 = 0x00000000
x4 = 0x00000000
x5 = 0x00000000
x6 = 0xaaaaaaaa
x7 = 0xaaaaaaaa
x8 = 0xaaaaaaaa
x9 = 0xaaaaaaaa
x10 = 0x00000800
x11 = 0xaaaaaaaa
x12 = 0xaaaaaaaa
x13 = 0xaaaaaaaa
x14 = 0xaaaaaaaa
x15 = 0x00000000
x16 = 0x00000000
x17 = 0x00000000
x18 = 0x00000000
x19 = 0x00000000
x20 = 0x00000000
x21 = 0x00000000
x22 = 0x00000000
x23 = 0x00000000
x24 = 0x00000000
x25 = 0x00000000
x26 = 0x00000000
x27 = 0x00000000
x28 = 0x00000000
x29 = 0x00000000
x30 = 0x00000000
x31 = 0x00000000
{x31, ..., x0} = 0x0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa00000800aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa000000000000000000000000000000000000100000000000
Quitting simulation.
